plugins/kubernetes-backend/src/service/KubernetesProxy.ts

```diff
@@ -138,7 +138,15 @@ export class KubernetesProxy {
         ws: true,
         secure: !originalCluster.skipTLSVerify,
         changeOrigin: true,
-        pathRewrite: { [`^${originalReq.baseUrl}`]: '' },
+        pathRewrite: async (path, req) => {
+          // Re-evaluate the cluster on each request, in case it has changed
+          const cluster = await this.getClusterForRequest(req);
+          const url = new URL(cluster.url);
+          return path.replace(
+            new RegExp(`^${originalReq.baseUrl}`),
```

Comment:
I think our original implementation may have suffered from a similar memoization-weakness. While we're here, does it work if we do something like this?
```suggestion
            new RegExp(`^${req.baseUrl}`),
```